### DISCUSSION TITLE
Unable to delete timestamp and source_type fields

### DISCUSSION THREAD

**Discussion by AmitJ1303** [vectordotdev/vector#19791] [Q&A]
Hi Team,
I tried to use del(.field1) as mentioned in VRL examples but was not able to delete timestamp and source_type fields. I am also validating JSON format in the .vrl files which seems correct.
Kindly suggest on how to remove these parameters. I am using vector version 0.32.0

**Comment by jszwedko** [comment]
Hey! `del(.field1)` is correct. Can you share the full configuration that you are trying?

**Comment by bmallya** [comment]
Hello @jszwedko, are these fields automatically added by vector? The event/log that was sent by an upstream instance of vector did not have these fields.

**Comment by AmitJ1303** [comment]
Just to update on configurations, we are running an instance at agent which collects logs from other applications through http, tcp, unix and sends it to other instance at collector through sink type vector. While receiving logs at agent we are not receiving source_type and timestamp but still it is appended in logs while sending it to collector. Please find below example,
JSON received at agent: 
{"SeverityText": "Trace","SeverityNumber":"5","name": "NetError","Body": "TEST MESSAGE 1","flags":"A","TraceId": "xxxnd-548","span_id": "","attributes":{ "thread.id":"9879", "code.function":"connect()","code.lineno":"90"},"resource": { "attributes":{"os.type":"linux","os.name":"fedora","device.lru_type":"372","device.lru_sub_type":"001"}}}
JSON sent by agent after processing:
{"Body":"TEST MESSAGE 1","SeverityNumber":"5","SeverityText":"Trace","TraceId":"xxxnd-548","attributes":{"code.function":"connect()","code.lineno":"90","thread.id":"9879"},"flags":"A","name":"NetError","resource":{"attributes":{"device.lru_sub_type":"001","device.lru_type":"372","os.name":"fedora","os.type":"linux"}},"source_type":"vector","span_id":"","timestamp":"2024-02-02T13:17:02.395694135Z"}

Do let us know how to exclude source_type, timestamp fields.

**Reply by jszwedko** [reply]
I'm having trouble following here. Could you please provide a complete example? `source_type` and `timestamp` are added by some sources but are removable via a `remap` transform using the `del()` function.

**Comment by AmitJ1303** [comment]
We have found the reason why it was not working. del() function is now working. We tried to add in collector instance and it worked. Previously we were modifying at agent before sending. I am closing this comment. Thanks for helping.
